portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -26,12 +32,18 @@
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
-  private ArrayList<String> comments = new ArrayList<String>( 
-    Arrays.asList());
+  private DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
   
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    String json = new Gson().toJson(this.comments);
+    ArrayList<String> comments = new ArrayList<>();
+    Query query = new Query("Comment");
```

Comment:
Once you have timestamp in your datastore then you add to the query to return to you the items in order.

Reply:
Done.